qemu/target-i386/translate.c

```diff
@@ -7425,14 +7425,11 @@ static target_ulong disas_insn(CPUX86State *env, DisasContext *s,
         /* For Intel SYSENTER is valid on 64-bit */
         if (CODE64(s) && env->cpuid_vendor1 != CPUID_VENDOR_INTEL_1)
             goto illegal_op;
-        if (!s->pe) {
-            gen_exception(s, EXCP0D_GPF, pc_start - s->cs_base);
-        } else {
```

Comment:
I admit it is purely based on my observation on how SYSCALL is handled. Also, during my tests, this condition was true, so there might be another issue explaining why sysenter_helper is not called.